### PR TITLE
Ported RxExchange to RxJava2 and added a new tests

### PR DESCRIPTION
**Pull request by tobias-bahls**
* Ported RxExchange to RxJava2
* Added an additional test

### CODE REVIEW

File: src/main/java/org/zalando/undertaking/RxExchange.java [hunk 1]
```diff
@@ -143,11 +145,18 @@ public void run() {
...
-        public void onNext(final HttpHandler handler) {
-            unsubscribe();
-            Connectors.executeRootHandler(handler, exchange);
+        public void onSubscribe(final Disposable d) {
+            this.disposable = d;
+        }
+
+        @Override
+        public void onSuccess(final HttpHandler httpHandler) {
+            if (!this.disposable.isDisposed()) {
```
Comment (cberg-zalando) [inline src/main/java/org/zalando/undertaking/RxExchange.java]: If `this` is not needed to identify the attribute, it should be left out. And apparently you're doing just that in the next line.

File: src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 2]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (cberg-zalando) [inline src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: What does this file do?

Comment (tobias-bahls) [inline src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Allows mocking final classes (see http://stackoverflow.com/questions/14292863/how-to-mock-a-final-class-with-mockito/14307615)

**Comment by cberg-zalando** [comment]
:+1:

**Pull request merged**